packages/flutter_tools/lib/src/commands/config.dart

```diff
@@ -128,6 +130,9 @@ class ConfigCommand extends FlutterCommand {
     if (argResults.wasParsed('clear-ios-signing-cert'))
       _updateConfig('ios-signing-cert', '');
 
+    if (argResults.wasParsed('build-dir'))
+      _updateConfig('build-dir', argResults['build-dir']);
```

Comment:
Is this the right place to validate that it is a relative path?

Reply:
Fixed and added test